### PR TITLE
CD 205. Adding spdlogs to casper-cpp-sdk

### PR DESCRIPTION
**Pull request by Piotrek321**

### CODE REVIEW

File: src/CasperClient.cpp [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 namespace Casper {
 
 /// Construct a new Casper Client object
 Client::Client(const std::string& address)
     : mAddress{address},
       mHttpConnector{mAddress},
-      mRpcClient{mHttpConnector, jsonrpccxx::version::v2} {}
+      mRpcClient{mHttpConnector, jsonrpccxx::version::v2} 
+      {
+        LogConfig log_config{.log_name = "Casper-SDK",
```
[the remark this answers is not in the document]
Comment (Piotrek321) [reply src/CasperClient.cpp:12]: done

File: src/include/ByteSerializers/CLValueByteSerializer.h [hunk 2]
```diff
@@ -38,7 +38,7 @@ struct CLValueByteSerializer : public BaseByteSerializer {
   void CLTypeToBytes(CBytes& sb, CLType innerType, CLTypeParsedRVA parsed) {
     // std::cout << "CLTypeToBytes1: " << std::endl;
     int type_idx = innerType.type.which();
-    std::cout << "CLTypeToBytes idx: " << type_idx << std::endl;
+    SPDLOG_DEBUG("CLTypeToBytes idx: ", type_idx);
```
[the remark this answers is not in the document]
Comment (Piotrek321) [reply src/include/ByteSerializers/CLValueByteSerializer.h:41]: done

File: src/include/Types/PublicKey.h [hunk 3]
```diff
@@ -133,14 +133,14 @@ struct PublicKey {
...
-        std::cout << "Bytes size: " << rawBytes.size() << std::endl;
+        SPDLOG_ERROR("rawBytes.size(): {} different than expectedPublicKeySize: {}", rawBytes.size(), expectedPublicKeySize);
 
         throw std::runtime_error(
             "Wrong public key format. Expected length is " +
             std::to_string(expectedPublicKeySize));
       }
     } catch (std::exception& e) {
-      std::cout << e.what() << std::endl;
+      SPDLOG_ERROR(e.what());
```
[the remark this answers is not in the document]
Comment (Piotrek321) [reply src/include/Types/PublicKey.h:143]: Im not sure about that... I added couple of info to exception thrown, but this log seems ok, exceptions have good explanation for me

File: src/include/Types/Secp256k1Key.cpp [hunk 4]
```diff
@@ -1,3 +1,4 @@
+#include <spdlog/spdlog.h>
```
[the remark this answers is not in the document]
Comment (Piotrek321) [reply src/include/Types/Secp256k1Key.cpp]: done

**Review by bartalke — APPROVED** [review]

**Pull request merged**